examples/step-74/step-74.cc

```diff
@@ -0,0 +1,1059 @@
+        const unsigned int n_q_points  = q_points.size();
+        const std::vector<double> &JxW = scratch_data.get_JxW_values();
+
+        std::vector<double> rhs(n_q_points);
+        rhs_function->value_list(q_points, rhs);
+
+        for (unsigned int point = 0; point < n_q_points; ++point)
+          for (unsigned int i = 0; i < fe_v.dofs_per_cell; ++i)
+            {
+              for (unsigned int j = 0; j < fe_v.dofs_per_cell; ++j)
+                copy_data.cell_matrix(i, j) +=
+                  diffusion_coefficient *     // nu
+                  fe_v.shape_grad(i, point) * // grad v_h
+                  fe_v.shape_grad(j, point) * // grad u_h
+                  JxW[point];                 // dx
+
+              copy_data.cell_rhs(i) += fe_v.shape_value(i, point) * // v_h
+                                       rhs[point] *                 // f
+                                       JxW[point];                  // dx
+            }
+      };
+
+    // This function assembles face integrals on the boundary.
+    const auto boundary_worker = [&](const auto &        cell,
+                                     const unsigned int &face_no,
+                                     auto &              scratch_data,
+                                     auto &              copy_data) {
+      const FEFaceValuesBase<dim> &fe_fv = scratch_data.reinit(cell, face_no);
+
+      const auto &       q_points      = scratch_data.get_quadrature_points();
+      const unsigned int n_q_points    = q_points.size();
+      const unsigned int dofs_per_cell = fe_fv.dofs_per_cell;
+
+      const std::vector<double> &        JxW = scratch_data.get_JxW_values();
+      const std::vector<Tensor<1, dim>> &normals =
+        scratch_data.get_normal_vectors();
+
+      std::vector<double> g(n_q_points);
+      exact_solution->value_list(q_points, g);
+
+      const double extent1 = cell->extent_in_direction(
+        GeometryInfo<dim>::unit_normal_direction[face_no]);
+      const double penalty = compute_penalty(degree, extent1, extent1);
+
+      for (unsigned int point = 0; point < n_q_points; ++point)
+        {
+          for (unsigned int i = 0; i < dofs_per_cell; ++i)
+            for (unsigned int j = 0; j < dofs_per_cell; ++j)
+              copy_data.cell_matrix(i, j) +=
+                (-diffusion_coefficient *        // - nu
+                   fe_fv.shape_value(i, point) * // v_h
+                   (fe_fv.shape_grad(j, point) * // (grad u_h .
+                    normals[point])              //  n)
+
+                 - diffusion_coefficient *         // - nu
+                     (fe_fv.shape_grad(i, point) * // (grad v_h .
+                      normals[point]) *            //  n)
+                     fe_fv.shape_value(j, point)   // u_h
+
+                 + diffusion_coefficient * penalty * // + nu sigma
+                     fe_fv.shape_value(i, point) *   // v_h
+                     fe_fv.shape_value(j, point)     // u_h
```

Comment:
In the following test:

https://github.com/dealii/dealii/blob/2b562ba0c55edd37a28613ae91419a2e39515b1b/tests/matrix_free/multigrid_dg_sip_01.cc#L271

we use `+(v, \nu \sigma * 2.0 * u)` is there a reason for this discrepancy?

Reply:
@zjiaqi2018 ?

Reply:
I assume @zjiaqi2018 has the factor of 2 inside the definition of the penalty term.

Reply:
@tjhei  @peterrum Because they use average instead of jump just like step-39. 
They define 
`average_value =
              (fe_eval.get_value(q) - fe_eval_neighbor.get_value(q)) * 0.5;`
but we define a jump which is
` (fe_eval.get_value(q) - fe_eval_neighbor.get_value(q)) ;`
and that's why they need to multiply by the factor 2.